### PR TITLE
Upgrade vanilla-cookieconsent to version 3.0.0.

### PR DESCRIPTION
**Pull request by EreMaijala**
This fixes navigation using tab when the cookie consent banner is displayed. Only minor changes to less/scss were made.

### CODE REVIEW

File: themes/bootstrap3/js/vendor/cookieconsent.umd.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
...e).getTime()>(n.expirationTime||0),e.T&&De(t.cookie.name)),e.T,(()=>{const e=g.o;for(const t of e.O){const n=e.P[t];if(n.readOnly||n.enabled&&e.i.mode===o){e.G.push(t);const o=e.X[t]||{};for(let n in o)e.Y[t].push(n)}}})(),e.T?t.mode===o&&(e.R=[...e.G]):(e.Y={...e.Y,...a},U([...e.B,...s])),e.Z={...e.Y}})();const i=Ue();if(!await xe())return!1;if($(null,r=Ge,be,Te),g.o.T&&he(r,Te),g.t.lazyHtmlGeneration||be(r,Te),n.autoShow&&!i&&Pe(!0),i)return ne(),te(s.le);n.mode===o&&ne(t.G)}var r},e.setCookieData=e=>{let t,o=e.value,n=e.mode,s=!1;const a=g.o;if('update'===n){a.h=t=Je('data');const e=typeof t==typeof o;if(e&&'object'==typeof t){!t&&(t={});for(let e in o)t[e]!==o[e]&&(t[e]=o[e],s=!0)}else!e&&t||t===o||(t=o,s=!0)}else t=o,s=!0;return s&&(a.h=t,a.p.data=t,Ae(!0)),s},e.setLanguage=async(e,t)=>{if(!Ce(e))return!1;const o=g.o;return!(e===we()&&!0!==t||!await xe(e)||(Se(e),o.D&&he(Ge,Te),o.N&&be(Ge,Te),Me(),0))},e.show=Pe,e.showPreferences=Re,e.validConsent=Ue,e.validCookie=e=>''!==He(e,!0)},'object'==typeof exports&&'undefined'!=typeof module?t(exports):'function'==typeof define&&define.amd?define(['exports'],t):t((e='undefined'!=typeof globalThis?globalThis:e||self).CookieConsent={});
```
[the remark this answers is not in the document]
Comment (EreMaijala) [reply themes/bootstrap3/js/vendor/cookieconsent.umd.js:7]: This is the old one.

**Comment by demiankatz** [comment]
@sturkel89, do you mind taking a look at this one? It's an upgrade to the library we use for cookie consent pop-ups (which are turned on with the `consent` setting in config.ini). Could you compare this latest version to the one in dev across the three themes and report any oddities you see?

(You'll may need to clear cookies between tests, since once you give consent, it is stored and the pop-up won't return).

Note that there's also an option (I believe in the footer) to adjust consents after the initial interaction.

**Review by sturkel89 — APPROVED** [review]
It's hard for me to see differences between test and dev for this PR, and appearance seems to be pretty much the same between themes (besides obvious color-coding of the green buttons in bootprint3). I see in the PR that test “fixes navigation using tab when the cookie consent banner is displayed” but I am not sure what that means. So, my review will focus on design issues that I found non-optimal or confusing.

### Cookie Pop-up
Simply looking at the layout, **I think the positioning of the text in the cookie banner should be left-aligned (with a little bit of padding)** vs. the way it currently resizes the padding depending on the width of the browser window. It looks strange with the large area of padding that appears in relatively wide window widths. (This positioning works the same way in all branches and in test and dev.) 

Examples of how the padding varies at 100% zoom:

**_Narrowest_**

image
---
**_A little wider_**

image
---
**_Even wider_**

image
---
**_Widest_**

image


### Cookie Settings lightbox
**Content**
In terms of content, I like how the Cookie Settings page offers details about the specific cookies, what they do, and when they expire. Has any of this content changed between test and dev? I didn’t do a word-for-word comparison.

image

**Design**
In terms of design, I see that the background of the **Analytics Cookies** slider is darker than the background of the **Essential Cookies** slider, and the Analytics one gets even darker when you switch from rejecting it to accepting it.

**_Optional rejected_**

image
---

**_Optional accepted_**

image


I guess the lighter display of the Essential Cookies slider background is intended to suggest that the option is grayed out (cannot be deselected). (I see that you get a NO symbol on mouseover of the Essential Cookies selection slider!) At first the darker shading in the slider for Optional Cookies confused me, but I guess this makes sense. **My only thought would be to replace the Essential Cookies slid-over checkmark image with a lozenge containing the word REQUIRED.** 

### Functionality
As far as I can tell, selecting either option is successful in preventing the cookie pop-up from reappearing later in the session. (I only did a bit more searching and browsing after accepting cookies.)

Please let me know what’s been changed from dev to test, if there’s anything else you want me to inspect. As it is, I don’t see any differences between test and dev.

**Comment by EreMaijala** [comment]
@sturkel89 Thanks for testing! Your results confirm that there are no side-effects from the update. Not seeing anything different is expected. I appreciate the comments on the look and feel, though I'm not sure I agree with the left-aligning of the banner. In any case the suggestions should probably be added to Jira so that they don't get lost, since they're not at focus here.

If you'd like to test the actual fix here, try the following:

1. Clear cookies so that you get the banner
2. Don't accept anything but leave the banner on screen
3. Open e.g. login dialog
4. Try to move from field to field with Tab
5. Expected results: In dev branch Tab doesn't work, in this branch Tab works.

**Comment by sturkel89** [comment]
I tested and see what you mean about tab not working on dev. Thanks for pointing out that difference. I would not have discovered this on my own!

I created a JIRA ticket re: my preference for re-doing how alignment works on the cookier banner: https://openlibraryfoundation.atlassian.net/browse/VUFIND-1663 .

As far as I'm concerned, this PR can be merged. Thanks!

**Review by demiankatz — APPROVED** [review]
Thanks, @sturkel89 and @EreMaijala!

**Pull request merged**